openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -205,6 +209,25 @@ def celebrations(self):
             'first_section': CourseEnrollmentCelebration.should_celebrate_first_section(self.enrollment_object),
         }
 
+    @property
+    def user_has_passing_grade(self):
+        course = get_course_by_id(self.course_key)
+        user_grade = CourseGradeFactory().read(self.request.user, course).percent
+
+        return user_grade >= course.lowest_passing_grade
+
+    @property
+    def course_completion_is_active(self):
+        return course_completion_is_active(self.course_key)
+
+    @property
+    def certificate_data(self):
+        course = get_course_with_access(self.request.user, 'load', self.course_key, check_if_enrolled=True)
```

Comment:
Have to also be careful with using `self.request.user` in this function because of masquerading. So really, you might not need to store the request as part of self anyway.